addons/point_of_sale/models/pos_config.py

```diff
@@ -355,6 +355,23 @@ def _check_payment_method_receivable_accounts(self):
                 _("You must configure an intermediary account for the payment methods: %s.") % method_names
             )
 
+    @api.constrains('module_pos_loyalty')
+    def _check_loyalties_product_availability(self):
+        if self.module_pos_loyalty and self.loyalty_id:
```

Comment:
more simpler and work with multi records
```python
def _check_loyalties_product_availability(self):
    archived_products = self.mapped('loyalty_id.reward_ids.point_product_id').filtered(
                        lambda p: not p.active).
   if archived_products:
       raise ValidationError(
                    _('The Discount Product of the following Loyalty Program are not available in the PoS.\n') + '\n'.join(archived_products.mapped('display_name') )
```

Reply:
Updated.
But I still prefer to show the id of the products as several products may have the same name (which was the case on the client database).
I also modify the message as the client may want to change to another product his loyalty.
Thank you for your proposition